src/AiLibrary.flix

```diff
@@ -25,18 +25,18 @@ namespace Flixball/AiLibrary {
         use RegionedRandom.nextInt64;
         MoveLogic((_, _) -> state -> region r {
             // use "0" if there's a bad encoding
-            let ran = RegionedRandom.newWithSeed(r, toInt64(state) |> Option.getWithDefault(0i64));
+            let ran = RegionedRandom.newWithSeed(r, fromState(state) |> Option.getWithDefault(0i64));
```

Comment:
The functions shouldn't really use this directly right? We'd add a function `fixState(ml: GameState -> Id -> s -> (Move, s)): Movelogic with State[s]` for the functional ai's and another one for the mutstate ones

Reply:
Right. I'll follow up with another PR to use the new function.